agents/temporal_difference.h

```diff
@@ -0,0 +1,27 @@
+#define LEARNING_RATE 0.02
+#define GAMMA 0.99
+#define MODEL_NAME "state_value.bin"
+
+#define CALC_STATE_NUM(x)                 \
+    {                                     \
+        x = 1;                            \
+        for (int i = 0; i < N_GRIDS; i++) \
+            x *= 3;                       \
+    }
+
+#define for_each_empty_grid(i, table) \
```

Comment:
Rename it to `for_each_empty_cell` and move it to `game.h`

Reply:
> **Tic-tac-toe**
> Tic-tac-toe is played on a three-by-three grid by two players, who alternately place the marks X and O in one of the nine spaces in the grid
> [wiki](https://en.wikipedia.org/wiki/Tic-tac-toe)

> **grid**
> a pattern or structure made from horizontal and vertical lines crossing each other to form squares
> [cambridge dictionary](https://dictionary.cambridge.org/zht/%E8%A9%9E%E5%85%B8/%E8%8B%B1%E8%AA%9E-%E6%BC%A2%E8%AA%9E-%E7%B9%81%E9%AB%94/grid)

In my opinion, we can call a game board a grid, but I'm not sure whether or not a cell on the board can be called a grid.
And I also think `N_GRIDS` is not a good name, it should be `N_CELLS`.